itensor/mps/mps.cc

```diff
@@ -1537,6 +1538,167 @@ inner(MPS const& psi, MPS const& phi) //Re[<psi|phi>]
     return re;
     }
 
+
+//----------------------------------------------------------------
+//
+//  CorrelationMatrix() implementation
+//
+
+// Convert a null range into a 1..N range.
+void fixRange(detail::RangeHelper<int>&r,int N)
+{
+    if (*r==*r.end()) 
+        r=range1(1,N);
+}
+
+//-------------------------------------------------------------------------------------------
+//
+//  Template implementation of correlationMatrix function for Real and Complex types,
+//
+template <class T> std::vector<std::vector<T>>
+correlationMatrixT(const MPS& _psi,
+                   const SiteSet& sites,
+                   const string& _op1,
+                   const string& _op2,
+                   detail::RangeHelper<int> site_range,
+                   Args const& args
+                  )
+{
+    assert(checkConsistent(_psi,sites));
+    bool isHermitian= args.getBool("isHermitian",true);
```

Comment:
Should this default to `true` or `false`?

I would vote for `false` to be on the safe side and avoid surprising behavior if the operators don't commute.

@emstoudenmire what do you think?

Reply:
Thanks Matt. I was thinking about this issue in the context of the Julia code too, since you raised it there too. 

What do you both think of this idea: calling the two operators A and B, if A is the Hermitian conjugate of B, then we can detect `isHermitian=true` automatically and do the more efficient algorithm, otherwise do the less efficient one. We could just put in a check like `norm(A - dag(swapPrime(B,0,1))) < 1E-10` or similar to check whether A is the conjugate of B. Since A and B are small (low-dimensional) operators, a numerical check like this ought to be reliable I think. If for some reason the check fails, all that will happen is a slightly slower algorithm will run but the output would still be correct.

Reply:
We could still let the user specifiy `isHermitian` to override this check when they are sure about it, and to ensure a certain level of performance, but then the default should be `false`.

Reply:
@emstoudenmire thanks for the comment, I didn't consider the idea of forming the operator and checking the Hermiticity of the tensor. That is a more general and automated solution than the one I brought up.

I think it is fine to check the Hermiticity up to some tolerance. I guess more specifically the operators only need to be Hermitian up to a scale factor, so we could check the Hermiticity of the normed operators:
```C++
norm(A / norm(A) - dag(swapPrime(B / norm(B), 0, 1))) < 1E-10
```